homeassistant/components/sensor/worldtidesinfo.py

```diff
@@ -0,0 +1,135 @@
+import requests
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=3600)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_LATITUDE): cv.string,
+    vol.Required(CONF_LONGITUDE): cv.string,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the WorldTidesInfo sensor."""
+    data = WorldTidesInfoData(hass)
+    name = "WorldTidesInfo"
+
+    global _RESOURCE
+    global LAT
+    global LON
+    global KEY
+
+    LAT = config.get(CONF_LATITUDE)
+    LON = config.get(CONF_LONGITUDE)
+    KEY = config.get(CONF_API_KEY)
+
+    STARTTIME = int(time.time())
+    _RESOURCE = 'https://www.worldtides.info/api?extremes&length=86400&start=%s&lat=%s&lon=%s&key=%s' % (
+        STARTTIME, LAT, LON, KEY)
+
+    try:
+        data.update()
+    except RunTimeError:
+        _LOGGER.error("Unable to connect fetch WorldTidesInfo data: %s")
+        return False
+
+    add_devices([WorldTidesInfoSensor(data, name)])
+
+
+class WorldTidesInfoSensor(Entity):
+    """Representation of a WorldTidesInfo sensor."""
+
+    def __init__(self, data, name):
+        """Initialize a WorldTidesInfo sensor."""
+        self.data = data
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self.data.data:
+            if "High" in str(self.data.data['extremes'][0]['type']):
+                tidetime = time.strftime('%I:%M %p', time.localtime(self.data.data['extremes'][0]['dt']))
+                return "High tide at %s" % (tidetime)
+            elif "Low" in str(self.data.data['extremes'][0]['type']):
+                tidetime = time.strftime('%I:%M %p', time.localtime(self.data.data['extremes'][1]['dt']))
+                return "Low tide at %s" % (tidetime)
+            else:
+                return STATE_UNKNOWN
+        else:
+            return STATE_UNKNOWN
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of this device."""
+        attr = {}
+        if "High" in str(self.data.data['extremes'][0]['type']):
+            attr['high_tide_time_utc'] = self.data.data['extremes'][0]['date']
+            attr['high_tide_height'] = self.data.data['extremes'][0]['height']
+            attr['low_tide_time_utc'] = self.data.data['extremes'][1]['date']
+            attr['low_tide_height'] = self.data.data['extremes'][1]['height']
+        elif "Low" in str(self.data.data['extremes'][0]['type']):
```

Comment:
This is a reversal of the "if" it switches sets high tide vs low tide depending on the order of the returned json